jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPart.java

```diff
@@ -631,20 +631,13 @@ public Content.Chunk read()
                     Content.Chunk chunk = part.getContent().read();
                     if (chunk == null || chunk instanceof Content.Chunk.Error)
                         yield chunk;
-                    if (chunk.isLast())
-                    {
-                        if (!chunk.hasRemaining())
-                        {
-                            chunk.release();
-                            chunk = Content.Chunk.EMPTY;
-                        }
-                        else
-                        {
-                            chunk = Content.Chunk.from(chunk.getByteBuffer(), false, chunk);
-                        }
-                        state = State.MIDDLE;
-                    }
-                    yield chunk;
+                    if (!chunk.isLast())
+                        yield chunk;
+                    state = State.MIDDLE;
+                    if (chunk.hasRemaining())
+                        yield Content.Chunk.from(chunk.getByteBuffer(), false, chunk);
+                    chunk.release();
+                    yield Content.Chunk.EMPTY;
```

Comment:
The from code already does the remaining and EMPTY/EOF checks:
```suggestion
                    yield Content.Chunk.from(chunk.getByteBuffer(), false, chunk);
```